src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java

```diff
@@ -0,0 +1,120 @@
+package edu.ucsb.cs156.gauchoride.controllers;
+
+import edu.ucsb.cs156.gauchoride.entities.Ride;
+import edu.ucsb.cs156.gauchoride.errors.EntityNotFoundException;
+import edu.ucsb.cs156.gauchoride.repositories.RideRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+
+@Api(description = "Ride Request")
+@RequestMapping("/api/ride_request")
+@RestController
+
+public class RideController extends ApiController {
+
+    @Autowired
+    RideRepository rideRepository;
+
+    @ApiOperation(value = "List all rides")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Ride> allRides() {
+        Iterable<Ride> rides = rideRepository.findAll();
+        return rides;
+    }
+
+    @ApiOperation(value = "Get a single ride")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public Ride getById(
+            @ApiParam("id") @RequestParam Long id) {
+        Ride ride = rideRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Ride.class, id));
+
+        return ride;
+    }
+
+    @ApiOperation(value = "Create a new ride")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public Ride postRide(
+        @ApiParam("day") @RequestParam String day,
+
+        @ApiParam("student") @RequestParam String student,
```

Comment:
This should probably not be a String, but rather the userid of the rider.

Remember that we want riders that login to be able to see their own rides.  That means the rides need to be connected to a user via their userid.

Reply:
And if it's an admin, this id can be specified, but if it's a rider, it should implicitly be the userid of the rider creating the ride.